src/server/components/upgrades.ts

```diff
@@ -3,17 +3,24 @@ import Component from '@/server/component';
 export default class Upgrades extends Component {
   public amount: number;
 
-  public speed = 0;
+  public speed: number;
 
-  public defense = 0;
+  public defense: number;
 
-  public energy = 0;
+  public energy: number;
 
-  public missile = 0;
+  public missile: number;
+
+  reset() {
```

Comment:
I consider the components as data sources, and all the logic of working with data is moved to other parts of the application (for example, in the systems). This removes the ambiguity: "where exactly should the reset method be implemented?"

Reply:
Should I just cutpaste the '= 0'? Which file could a helper like that live in. What about helpers.js?

Reply:
I think you can create the `maintenance/players/upgrades.ts` file and implement the methods to work with upgrades there. There is also a file `/commands/upgrades.ts` where the information about player upgrades changes.

Reply:
The cutpaste is brought back for now to match what was done previously. Do you mean to add a new RESET_UPGRADES event or similar, and implement a handler for it?

Reply:
Yep, something like PLAYERS_UPGRADES_RESET.